File: https:/github.com/abhinavtripathi17/DSA-Leetcode.git/169-MajorityElement/169-MajorityElement.cpp
// Last updated: 8/4/2026, 7:19:51 PM
class Solution {
public:
    int majorityElement(vector<int>& nums) {
        int n = nums.size();
        int maj = nums[0];
        int cnt = 1;

        int i = 1;
        while(i < n){
            if(maj != nums[i]){
                cnt--;
                if(cnt == 0){
                    maj = nums[i];
                    cnt = 1;
                }
            }
            else{
                cnt++;
            }
            i++;
        }
        return maj;
    }
};

File: https:/github.com/abhinavtripathi17/DSA-Leetcode.git/169-MajorityElement/169-MajorityElement.java
// Last updated: 8/4/2026, 7:19:58 PM
class Solution {
    public int majorityElement(int[] nums) {
        int n = nums.length;
        int var = nums[0];
        int count = 1;

        for(int i = 1 ; i < n ; i++){
            if(nums[i] == var) count++;

            else count--;

            if(count == 0){
                var = nums[i];
                count = 1;
            }
        }
        return var;
    }
}